invenio_github/api.py

```diff
@@ -194,101 +208,178 @@ def _sync_hooks(self, repos, asynchronous=True):
         if ghrepo:
-            hooks = (h for h in ghrepo.hooks()
-                     if h.config.get('url', '') == self.webhook_url)
+            hooks = (
+                h for h in ghrepo.hooks() if h.config.get("url", "") == self.webhook_url
+            )
             hook = next(hooks, None)
             if not hook or hook.delete():
-                Repository.disable(user_id=self.user_id,
-                                   github_id=repo_id,
-                                   name=name)
+                Repository.disable(repo)
                 return True
         return False
 
+    def get_repository_releases(self, repo_name="", repo=None):
+        """Retrieve repository releases."""
+        if not (repo or repo_name):
+            raise ValueError("At least one of (repo, repo_name) is required")
+
+        if not repo:
+            repo = Repository.get(name=repo_name)
+
+        check_repo_access_permissions(repo, self.user_id, repo.github_id, repo_name)
+
+        # Retrieve releases and sort them by creation date
+        release_objects = sorted(repo.releases.all(), key=lambda r: r.created)
+
+        return release_objects
+
+    def get_user_repositories(self):
+        """Retrieves user repositories."""
+        token = self.session_token
+        if token:
+            extra_data = self.account.extra_data
+        repos = extra_data.get("repos", [])
+        if repos:
+            # 'Enhance' our repos dict, from our database model
+            db_repos = Repository.query.filter(
+                Repository.github_id.in_([int(k) for k in repos.keys()]),
+            ).all()
+            for repo in db_repos:
+                repos[str(repo.github_id)]["instance"] = repo
+                repos[str(repo.github_id)]["latest"] = GitHubRelease(
+                    repo.latest_release()
+                )
+        return repos
+
+    def get_last_sync_time(self):
+        """Retrieves the last sync delta time from github's client extra data.
+
+        Time is computed as the delta between now and the last sync time.
+        """
+        if not self.account.extra_data.get("last_sync"):
+            raise RemoteAccountDataNotSet(
+                self.user_id, "Last sync data is not set for user (remote data)."
+            )
+
+        extra_data = self.account.extra_data
+        last_sync = humanize.naturaltime(
+            (utcnow() - parse_timestamp(extra_data["last_sync"]))
+        )
+        return last_sync
+
+    def get_repository(self, repo_name):
+        """Retrieves one repository.
+
+        Checks for access permission.
+        """
+        # Might raise a ResultNotFound
+        repo_instance = Repository.get(name=repo_name)
+
+        # Might raise a RepositoryAccessError
+        check_repo_access_permissions(
+            repo_instance, self.user_id, repo_instance.github_id, repo_name
+        )
+
+        return repo_instance
+
     @classmethod
     def _dev_api(cls):
```

Comment:
These three class methods were not tested yet by me. 

It seems only `revoke_token` is used by a task